ClosedXML/Excel/XLWorksheets.cs

```diff
@@ -160,9 +160,14 @@ public IXLWorksheet Add(DataTable dataTable)
         }
 
         public IXLWorksheet Add(DataTable dataTable, String sheetName)
+        {
+            return Add(dataTable, sheetName, TableNameGenerator.GetNewTableName(_workbook));
```

Comment:
It uses the new table name generator class and provides a generated table name, as suggested in the ticket.